src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncEHTests.cs

```diff
@@ -2104,5 +2104,148 @@ static void Main()
 ";
             CompileAndVerify(source, expected);
         }
+
+        [Fact, WorkItem(67091, "https://github.com/dotnet/roslyn/issues/67091")]
+        public void NestedCatch_DuplicateLocal()
+        {
+            var source = """
+                using System;
+                using System.Threading.Tasks;
+                using static System.Console;
+
+                class C
+                {
+                    static async Task Main()
+                    {
+                        await new C().M1();
+                    }
+
+                    int _counter;
+
+                    bool F(Exception ex)
+                    {
+                        _counter++;
+                        WriteLine($"F: {_counter} {ex.Message}");
+                        return _counter == 2;
+                    }
+
+                    async Task M1()
+                    {
+                        try
+                        {
+                            throw new Exception("M1");
+                        }
+                        catch (Exception ex) when (F(ex))
+                        {
+                            await M2(ex);
+                        }
+                        catch (Exception ex) when (F(ex))
+                        {
+                            try
+                            {
+                                throw new Exception("M1-catch2");
+                            }
+                            catch 
+                            {
+                                await M2(ex);
+                            }
+                        }
+                    }
+
+                    async Task M2(Exception ex)
```

Comment:
>async Task M2(Exception ex) [](http://example.com/codeflow?start=20&length=27)

It is hard to judge from the output about call site of this method. Consider explicitly passing some kind of call site id at each call site and printing it as well.